src/components/Tool/GoogleImport/GoogleImport.tsx

```diff
@@ -0,0 +1,553 @@
+  },
+  headerTitle: {
+    fontSize: '1rem!important',
+  },
+}));
+
+const googleImportSchema = yup.object({
+  tagsForAllList: yup.array().of(yup.string()).default([]),
+  override: yup.string().required(),
+  importAll: yup.string().required(),
+  groupTags: yup.object(),
+  groups: yup.array().of(yup.string()).default([]),
+});
+
+type Attributes = yup.InferType<typeof googleImportSchema>;
+
+interface Props {
+  accountListId: string;
+}
+
+const GoogleImport: React.FC<Props> = ({ accountListId }: Props) => {
+  const { classes } = useStyles();
+  const { appName } = useGetAppSettings();
+  const { t } = useTranslation();
+  const [showModal, setShowModal] = useState(false);
+  const [showConfirmAllModal, setShowConfirmAllModal] = useState(false);
+  const [redirecting, setRedirecting] = useState(false);
+  const [selectedAccount, setSelectedAccount] = useState('');
+
+  const { data: contactTagsList, loading: contactTagsListLoading } =
+    useGetContactTagListQuery({
+      variables: {
+        accountListId,
+      },
+    });
+
+  const { data, loading } = useGoogleContactGroupsQuery();
+  const googleAccounts = data?.googleAccounts;
+
+  const account = useMemo(() => {
+    return googleAccounts?.find((account) => account?.id === selectedAccount);
+  }, [googleAccounts, selectedAccount]);
+
+  const initialValues = useMemo(() => {
+    const initialGroupTags =
+      account?.contactGroups.reduce((acc, group) => {
+        if (group?.id) {acc[group?.id] = [group?.tag];}
+        return acc;
+      }, {}) || {};
+    return {
+      tagsForAllList: [],
+      override: 'false',
+      importAll: 'false',
+      groupTags: initialGroupTags,
+      groups: [] as string[],
+    };
+  }, [account]);
+
+  useEffect(() => {
+    googleAccounts?.length && setSelectedAccount(googleAccounts[0]?.id || '');
+  }, [googleAccounts]);
+
+  const handleCloseModal = () => {
+    setShowModal(false);
+    window.location.href = `${process.env.SITE_URL}/accountLists/${accountListId}/tools`;
+    setRedirecting(true);
+  };
+
+  const onSubmit = async (
+    attributes: Attributes,
+    { resetForm },
+  ): Promise<void> => {
+    resetForm();
+    alert('import mutation TODO');
+    setShowModal(true);
+  };
+
+  return (
+    <OuterBox>
+      {redirecting && (
+        <LoadingSpinner firstLoad={true} data-testid="LoadingSpinner" />
+      )}
+      <ContainerBox container>
+        <>
+          <Grid item xs={12}>
+            <Typography variant="h4">{t('Import from Google')}</Typography>
+            <Divider className={classes.divider} />
+          </Grid>
+          {loading && !data && (
+            <LoadingSpinner firstLoad={true} data-testid="LoadingSpinner" />
+          )}
+
+          {!loading && data && (
+            <>
+              {!data?.googleAccounts.length && <NoData tool="googleImport" />}
+              {!!data?.googleAccounts.length && (
+                <>
+                  {data?.googleAccounts.length > 1 && (
```

Comment:
This will always be true, right? Since we're already checking `!!data?.googleAccounts.length`?

Reply:
This is checking to see if there is more than 1 and that determines if we show a `Select` so they can choose between accounts.

Reply:
Sorry, yes you're correct.